File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/events/specs/GivenARandomBroadcast/WhenReceivingTheBroadcast.java
package com.lasthopesoftware.bluewater.client.library.events.specs.GivenARandomBroadcast;

import android.content.Intent;
import android.support.v4.content.LocalBroadcastManager;

import com.lasthopesoftware.bluewater.client.library.BrowseLibraryActivity;
import com.lasthopesoftware.specs.AndroidContext;

import org.junit.Test;
import org.robolectric.Robolectric;
import org.robolectric.android.controller.ActivityController;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

public class WhenReceivingTheBroadcast extends AndroidContext {

	private static final ActivityController<BrowseLibraryActivity> activityController = Robolectric.buildActivity(BrowseLibraryActivity.class).create().start().resume();

	@Override
	public void before() {
		final LocalBroadcastManager localBroadcastManager = LocalBroadcastManager.getInstance(activityController.get());

		final Intent broadcastIntent = new Intent("absr4");
		localBroadcastManager.sendBroadcast(broadcastIntent);
	}

	@Test
	public void thenTheActivityIsNotFinished() {
		assertThat(activityController.get().isFinishing()).isFalse();
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/browsing/library/events/GivenARandomBroadcast/WhenReceivingTheBroadcast.kt
package com.lasthopesoftware.bluewater.client.browsing.library.events.GivenARandomBroadcast

import com.lasthopesoftware.AndroidContext
import com.lasthopesoftware.bluewater.client.browsing.BrowserEntryActivity
import com.lasthopesoftware.bluewater.shared.messages.application.ApplicationMessage
import com.lasthopesoftware.bluewater.shared.messages.application.ApplicationMessageBus.Companion.getApplicationMessageBus
import org.assertj.core.api.AssertionsForClassTypes.assertThat
import org.junit.Test
import org.robolectric.Robolectric

class WhenReceivingTheBroadcast : AndroidContext() {

	companion object {
		private val activityController by lazy {
			Robolectric.buildActivity(BrowserEntryActivity::class.java).create().start().resume()
		}
	}

	override fun before() {
		System.setProperty("javax.net.ssl.trustStoreType", "JKS")
		activityController.get().getApplicationMessageBus().sendMessage(object : ApplicationMessage{})
	}

	@Test
	fun thenTheActivityIsNotFinished() {
		assertThat(activityController.get().isFinishing).isFalse
	}
}
